src/prefect/__init__.py

```diff
@@ -45,80 +30,95 @@
         suspend_flow_run,
     )
 
-_slots: dict[str, Any] = {
-    "__version_info__": __version_info__,
-    "__version__": __version__,
-    "__module_path__": __module_path__,
-    "__development_base_path__": __development_base_path__,
-    "__ui_static_subpath__": __ui_static_subpath__,
-    "__ui_static_path__": __ui_static_path__,
-}
+    __spec__: ModuleSpec
+
+    # Versioneer provides version information as dictionaries
+    # with these keys
+    class VersionInfo(TypedDict("_FullRevisionId", {"full-revisionid": str})):
+        version: str
+        dirty: Optional[bool]
+        error: Optional[str]
+        date: Optional[str]
+
+
+__version_info__: "VersionInfo" = cast("VersionInfo", _version.get_versions())
+__version__ = __version_info__["version"]
 
-_public_api: dict[str, tuple[str, str]] = {
+# The absolute path to this module
+__module_path__: pathlib.Path = pathlib.Path(__file__).parent
+# The absolute path to the root of the repository, only valid for use during development
+__development_base_path__: pathlib.Path = __module_path__.parents[1]
+
+# The absolute path to the built UI within the Python module, used by
+# `prefect server start` to serve a dynamic build of the UI
+__ui_static_subpath__: pathlib.Path = __module_path__ / "server" / "ui_build"
+
+# The absolute path to the built UI within the Python module
+__ui_static_path__: pathlib.Path = __module_path__ / "server" / "ui"
+
+del _version, pathlib
+
+_public_api: dict[str, tuple[Optional[str], str]] = {
     "allow_failure": (__spec__.parent, ".main"),
+    "aserve": (__spec__.parent, ".main"),
+    "deploy": (__spec__.parent, ".main"),
     "flow": (__spec__.parent, ".main"),
     "Flow": (__spec__.parent, ".main"),
     "get_client": (__spec__.parent, ".main"),
     "get_run_logger": (__spec__.parent, ".main"),
+    "pause_flow_run": (__spec__.parent, ".main"),
+    "resume_flow_run": (__spec__.parent, ".main"),
+    "serve": (__spec__.parent, ".main"),
     "State": (__spec__.parent, ".main"),
+    "suspend_flow_run": (__spec__.parent, ".main"),
     "tags": (__spec__.parent, ".main"),
     "task": (__spec__.parent, ".main"),
     "Task": (__spec__.parent, ".main"),
     "Transaction": (__spec__.parent, ".main"),
     "unmapped": (__spec__.parent, ".main"),
-    "serve": (__spec__.parent, ".main"),
-    "aserve": (__spec__.parent, ".main"),
-    "deploy": (__spec__.parent, ".main"),
-    "pause_flow_run": (__spec__.parent, ".main"),
-    "resume_flow_run": (__spec__.parent, ".main"),
-    "suspend_flow_run": (__spec__.parent, ".main"),
 }
 
 # Declare API for type-checkers
 __all__ = [
+    "__development_base_path__",
+    "__module_path__",
+    "__ui_static_path__",
+    "__ui_static_subpath__",
+    "__version__",
+    "__version_info__",
```

Comment:
Apart from `__version__`, the other custom dunder attributes are intended for internal use only, so we can leave them out of `__all__`.
```suggestion
    "__version__",
```